### ISSUE TITLE
Use the original dense CVPR network (no different homeomorphism) for training

### ISSUE THREAD

**Issue by SLKaMiHi** [voxelmorph/voxelmorph#331]
**Task** (what are you trying to do/register?)
I want to use the original dense CVPR network (no different homeomorphism) for training.What parameters need to be changed?

**Comment by adalca** [comment]
@SLKaMiHi The very original code is in the legacy branch, but we don't support it actively. TO get a model very similar to the original CVPR2018 paper, you can use our default train.py with int-steps set to 0, I believe. @ahoopes can you confirm?

**Comment by SLKaMiHi** [comment]
Hi：
I want to train the model of MICCAI 2018. What changes do I need to make. What parameters need to be adjusted?
Thank you.

**Comment by ahoopes** [comment]
Correct, setting `--int-steps 0` will disable diffeomorphisms

**Comment by ahoopes** [comment]
You can enable the deformation probabilities with the `--use-probs` flag, but to mimic the loss weights used in the MICCAI paper, you should also set `--legacy-image-sigma 0.02 --lambda 1.0`.

**Comment by SLKaMiHi** [comment]
I'm really sorry that I didn't express it clearly. I want to use the model training data of MICCAI2018. What kind of modification should be made?


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "voxelmorph/voxelmorph"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月15日(星期四) 凌晨3:50
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [voxelmorph/voxelmorph] Use the original dense CVPR network (no different homeomorphism) for training (#331)





 
And it's a bit difficult to parse that last message. There must have been a problem converting from email, but it sounds like you want to train the MICCAI2018-style model? You can enable the deformation probabilities with the --use-probs flag, but to mimic the loss weights used in the MICCAI paper, you should also set --legacy-image-sigma 0.02 --lambda 1.0.

**Comment by SLKaMiHi** [comment]
Hi：
I have a little problem at runtime. Is there an error in the code at the red line.I think it is log_sigma = y_pred[..., 0:ndims].
Thank you！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "voxelmorph/voxelmorph"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月15日(星期四) 凌晨3:50
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [voxelmorph/voxelmorph] Use the original dense CVPR network (no different homeomorphism) for training (#331)





 
And it's a bit difficult to parse that last message. There must have been a problem converting from email, but it sounds like you want to train the MICCAI2018-style model? You can enable the deformation probabilities with the --use-probs flag, but to mimic the loss weights used in the MICCAI paper, you should also set --legacy-image-sigma 0.02 --lambda 1.0.

**Comment by ahoopes** [comment]
Can you provide a bit more information about the error you're getting? And I'm not totally sure about the MICCAI2018 training data availability - @adalca will have to answer that one

**Comment by SLKaMiHi** [comment]
Hi:


MICCAI 2018.scan-to-scan.
The size of my data is 96*112*96. Using standard processing, using freesufer affine alignment and normalizing to 0-1.
I used a variety of hyperparameter settings during training.mse: (lambda=0.01,lambda_weight=10,image_sigma=0.02).NCC(lambda=1,lambda_weight=25,image_sigma=0.02) and so on.use_prob=True.int_step=7.


I am using the latest version of the code downloaded from the official website.
I used a variety of hyperparameter settings during training. But Dice always drops to around 0.41. I don't know where the problem.



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "voxelmorph/voxelmorph"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年8月10日(星期二) 凌晨4:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [voxelmorph/voxelmorph] Use the original dense CVPR network (no different homeomorphism) for training (#331)





 
Can you provide a bit more information about the error you're getting? And I'm not totally sure about the MICCAI2018 training data availability - @adalca will have to answer that one
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by SLKaMiHi** [comment]
MICCAI 2018.scan-to-scan.
The size of my data is 96*112*96. Using standard processing, using freesufer affine alignment and normalizing to 0-1.
I used a variety of hyperparameter settings during training.mse: (lambda=0.01,lambda_weight=10,image_sigma=0.02).NCC(lambda=1,lambda_weight=25,image_sigma=0.02) and so on.use_prob=True.int_step=7.

I am using the latest version of the code downloaded from the official website.
I used a variety of hyperparameter settings during training. But Dice always drops to around 0.41. I don't know where the problem.

**Comment by SLKaMiHi** [comment]
> Can you provide a bit more information about the error you're getting? And I'm not totally sure about the MICCAI2018 training data availability - @adalca will have to answer that one

MICCAI 2018.scan-to-scan.
The size of my data is 96*112*96. Using standard processing, using freesufer affine alignment and normalizing to 0-1.
I used a variety of hyperparameter settings during training.mse: (lambda=0.01,lambda_weight=10,image_sigma=0.02).NCC(lambda=1,lambda_weight=25,image_sigma=0.02) and so on.use_prob=True.int_step=7.

I am using the latest version of the code downloaded from the official website.
I used a variety of hyperparameter settings during training. But Dice always drops to around 0.41. I don't know where the problem.
